brute-force-connect/src/main/java/com/bakdata/kafka/BruteForceConverter.java

```diff
@@ -105,13 +107,23 @@ public void configure(final Map<String, ?> configs, final boolean isKey) {
                 BruteForceConverter.class.getSimpleName() + " only supports converting to connect data");
     }
 
+    /**
+     * @since 1.1.0
+     * @deprecated Use {@link Converter#toConnectData(String, Headers, byte[])}
+     */
     @Override
+    @Deprecated(since = "1.1.0")
     public SchemaAndValue toConnectData(final String topic, final byte[] value) {
+        return this.toConnectData(topic, new RecordHeaders(), value);
+    }
+
+    @Override
+    public SchemaAndValue toConnectData(final String topic, final Headers headers, final byte[] value) {
         Objects.requireNonNull(this.converters, "You need to configure the converter first");
         for (final Converter converter : this.converters) {
             final Class<? extends Converter> clazz = converter.getClass();
             try {
-                final SchemaAndValue schemaAndValue = converter.toConnectData(topic, value);
+                final SchemaAndValue schemaAndValue = converter.toConnectData(topic, headers, value);
```

Comment:
Could we run into a recursive problem here if `BruteForceConverter` is in the list of converters and should we check for that specifically?

Reply:
It is not part of the list